### ISSUE TITLE
Unexpected indentation error in functions returned from ternary operator 

### ISSUE THREAD

**Issue by jtraub** [eslint/eslint#8775]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 6.9.5
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

```
module.exports = {
  "extends": "airbnb",
  "parser": "babel-eslint",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "root": true,
  "env": {
    "browser": true
  },
  "rules": {
    "curly": ["error", "all"],
    "arrow-parens": ["error", "as-needed"],
  },
  "settings": {
    "import/resolver": {
      "webpack": {
        "config": "webpack.config.js",
      }
    }
  },
};
```

**What did you do? Please include the actual source code causing the issue.**

```js


function func1() {
  // do some stuff
}

function func2() {
  // do some stuff
}

function test(inp) {
  return inp
    ? () => {
        func2();
        return func1;
      }
    : () => {
        func1();
        return func2;
      };
}

test();




```

**What did you expect to happen?**

no eslint errors

**What actually happened? Please include the actual, raw output from ESLint.**

eslint test-indent.js 

/tmp/eslint-problem/test-indent.js
  13:9  error  Expected indentation of 6 spaces but found 8  indent
  14:9  error  Expected indentation of 6 spaces but found 8  indent
  15:7  error  Expected indentation of 4 spaces but found 6  indent
  17:9  error  Expected indentation of 6 spaces but found 8  indent
  18:9  error  Expected indentation of 6 spaces but found 8  indent
  19:7  error  Expected indentation of 4 spaces but found 6  indent

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended -- the `indent` rule enforces that the end of the function has the same alignment as the *line* that starts the function, not the start of the function itself.

As a similar example, this is why the rule considers this code to be indented correctly:

```js
foo(function() {
  bar();
});
```

rather than this code:

```js
foo(function() {
      bar();
    });
```

**Comment by jtraub** [comment]
Thanks @not-an-aardvark for the explanation.

The code above was produced by prettier. Related issue with some meaningful  discussion and links is https://github.com/prettier/prettier/issues/1271
